external-dns.yaml

```diff
@@ -63,8 +63,53 @@ update:
     tag-filter: v
 
 test:
+  environment:
+    contents:
+      packages:
+        - etcd
+        - bind-tools
+        - coredns
   pipeline:
-    # AUTOGENERATED
-    - runs: |
+    - name: Basic binary tests
+      runs: |
         external-dns --version
         external-dns --help
+    - name: Functional verification
+      runs: |
+        # Start a local etcd instance
+        etcd --listen-client-urls http://127.0.0.1:2379 --advertise-client-urls http://127.0.0.1:2379 &
+        etcd_pid=$!
+        sleep 2
+
+        etcdctl put /skydns/org/example/foo '{"host":"42.42.42.42"}'
+
+        # Start CoreDNS with etcd backend
+        cat > Corefile <<EOF
+        .:53 {
+            errors
+            health
+            etcd {
+                path /skydns
+                endpoint http://127.0.0.1:2379
+            }
+            forward . /etc/resolv.conf
+            cache
+        }
+        EOF
+        coredns -conf Corefile &
+        coredns_pid=$!
+        sleep 2
+
+        external-dns --provider coredns --log-level debug --etcd-endpoints http://127.0.0.1:2379 &
+        external_dns_pid=$!
+        sleep 2
+
+        dig @127.0.0.1 foo.example.org +short
+        if [ "$(dig @127.0.0.1 foo.example.org +short)" != "42.42.42.42" ]; then
+          echo "DNS resolution failed"
+          exit 1
+        fi
```

Comment:
`etcdctl put /skydns/org/example/foo '{"host":"42.42.42.42"}'`
manually populates the etcd key used by CoreDNS's etcd backend. Then CoreDNS serves that record. `external-dns` plays no part here.